virtme/commands/run.py

```diff
@@ -162,6 +165,13 @@ def quote_karg(arg):
     else:
         return arg
 
+def random_mac():
+    mac = [ 0x00, 0x16, 0x3e,
+            random.randint(0x00, 0x7f),
+            random.randint(0x00, 0xff),
+            random.randint(0x00, 0xff) ]
+    return ':'.join(map(lambda x: "%02x" % x, mac))
+
```

Comment:
Is there nothing in the standard library that does this?

That seems to be generating a random MAC owned by Xensource.  How about generating a real random MAC with the locally generated flag set?  Also, this should use os.urandom() or whatever the modern API for a secure random number is.

Reply:
Also it's not a good idea to use completely random mac addresses, because you can accidentally generate a MAC address with multicast bit set and wonder what the heck is happening.  See https://en.wikipedia.org/wiki/MAC_address#Universal_vs._local for a bit of a discussion; I'll suggest an alternative approach.